internal/aks/check.go

```diff
@@ -0,0 +1,180 @@
+		"koreacentral":       "koreacentral",
+		"koreasouth":         "koreacentral",
+		"northcentralus":     "northcentralus",
+		"northeurope":        "northeurope",
+		"southafricanorth":   "southafricanorth",
+		"southafricawest":    "southafricanorth",
+		"southcentralus":     "southcentralus",
+		"southeastasia":      "southeastasia",
+		"southindia":         "centralindia",
+		"uksouth":            "uksouth",
+		"ukwest":             "ukwest",
+		"westcentralus":      "eastus",
+		"westeurope":         "westeurope",
+		"westindia":          "centralindia",
+		"westus":             "westus",
+		"westus2":            "westus2",
+		"norwayeast":         "norwayeast",
+		"norwaywest":         "norwayeast",
+		"switzerlandnorth":   "switzerlandnorth",
+		"switzerlandwest":    "switzerlandwest",
+		"uaenorth":           "uaenorth",
+		"germanywestcentral": "germanywestcentral",
+		"germanynorth":       "germanywestcentral",
+		"uaecentral":         "uaecentral",
+		"eastus2euap":        "eastus2euap",
+		"brazilsoutheast":    "brazilsoutheast",
+		// mapping for azure china cloud
+		"chinaeast":   "chinaeast2",
+		"chinaeast2":  "chinaeast2",
+		"chinanorth":  "chinaeast2",
+		"chinanorth2": "chinaeast2",
+	}
+
+	workspaceRegion, ok := regionToOmsRegionMap[spec.ResourceLocation]
+	if !ok {
+		return "", fmt.Errorf("region %s not supported for Log Analytics workspace", spec.ResourceLocation)
+	}
+	workspaceRegionCode, ok := locationToOmsRegionCodeMap[workspaceRegion]
+	if !ok {
+		return "", fmt.Errorf("region %s not supported for Log Analytics workspace", workspaceRegion)
+	}
+
+	workspaceResourceGroup := to.String(spec.LogAnalyticsWorkspaceGroup)
+	if workspaceResourceGroup == "" {
+		workspaceResourceGroup = spec.ResourceGroup
+	}
+
+	workspaceName := to.String(spec.LogAnalyticsWorkspaceName)
+	if workspaceName == "" {
+		workspaceName = fmt.Sprintf("%s-%s", spec.ResourceGroup, workspaceRegionCode)
+	}
+	// workspaceName string length can be only 63
+	if len(workspaceName) > workspaceLength {
+		workspaceName = generateUniqueLogWorkspace(workspaceName)
+	}
+
+	if gotRet, gotErr := client.Get(ctx, workspaceResourceGroup, workspaceName); gotErr == nil {
+		return *gotRet.ID, nil
+	}
+
+	logrus.Infof("Create Azure Log Analytics Workspace %q on Resource Group %q", workspaceName, workspaceResourceGroup)
+
+	asyncRet, asyncErr := client.CreateOrUpdate(ctx, workspaceResourceGroup, workspaceName, operationalinsights.Workspace{
+		Location: to.StringPtr(workspaceRegion),
+		WorkspaceProperties: &operationalinsights.WorkspaceProperties{
+			Sku: &operationalinsights.WorkspaceSku{
+				Name: operationalinsights.WorkspaceSkuNameEnumStandalone,
+			},
+		},
+	})
+	if asyncErr != nil {
+		return "", asyncErr
+	}
+
+	err = wait.Poll(5*time.Second, 30*time.Second, func() (bool, error) {
+		ret, err := asyncRet.Result(*client)
+		if err != nil {
+			return false, err
+		}
+
+		workspaceID = *ret.ID
+		return true, nil
+	})
+	return
+}
```

Comment:
Does this do something similar that we could leverage https://github.com/Azure/aks-engine/blob/master/pkg/armhelpers/loganalytics.go#L17?

Reply:
It seems that there is nothing what we could include into our bits.